### PR TITLE
Fix TestVirtual_memory #552

### PR DESCRIPTION
**Pull request by elboulangero**
Based on https://stackoverflow.com/a/30772733/776208

Signed-off-by: Arnaud Rebillout <arnaud.rebillout@collabora.com>

### CODE REVIEW

File: mem/mem_test.go [hunk 1]
```diff
@@ -26,8 +26,8 @@ func TestVirtual_memory(t *testing.T) {
 	assert.True(t, v.Available > 0)
 	assert.True(t, v.Used > 0)
 
-	assert.Equal(t, v.Total, v.Available+v.Used,
-		"Total should be computable from available + used: %v", v)
+	assert.Equal(t, v.Total, v.Used + v.Free + v.Buffers, v.Cached,
```
Comment (Lomanic) [inline mem/mem_test.go]: `s/v.Buffers, v.Cached/v.Buffers + v.Cached/`, but tested OK on Linux otherwise.

Comment (shirou) [reply mem/mem_test.go]: Good catch. I will fix after merge.

**Comment by shirou** [comment]
Thank you so much!

File: mem/mem_test.go (hunk 1 above)
Comment (Lomanic) [reply mem/mem_test.go]: This test is now broken on Windows (it wasn't before this merge), because v.Free+v.Buffers+v.Cached equals 0.

Comment (elboulangero) [reply mem/mem_test.go]: Oh sorry, I have no idea how this thing works on Windows. Looks like the formula to get `v.Total` varies with the OS...

Comment (shirou) [reply mem/mem_test.go]: So, is this test itself wrong because platform dependent? if so, we'd better to delete this test.

Comment (elboulangero) [reply mem/mem_test.go]: I believe that the fix I provided is the right thing for Linux. I have no idea about Window. Not even talking about MacOS :) Looks like something platform dependent indeed.

Comment (Lomanic) [reply mem/mem_test.go]: I have a PR ready that makes the test branch out against `runtime.GOOS` so the test passes on Windows (that's not the only issue for this test on Windows), I will submit it for review in the upcoming days. 

I don't even have a darwin machine anymore to check if the test passes on this platform.

**Pull request merged**